scipy/io/wavfile.py

```diff
@@ -27,6 +28,39 @@ class WavFileWarning(UserWarning):
     pass
 
 
+class SeekEmulatingReader:
+    def __init__(self, reader):
+        self.reader = reader
+        self.pos = 0
+
+    def read(self, size=-1, /):
+        data = self.reader.read(size)
+        self.pos += len(data)
+        return data
+    
+    def seek(self, offset, whence=os.SEEK_SET, /):
+        match whence:
+            case os.SEEK_SET if offset == self.pos:
+                pass # advance to current position (no-op)
+            case os.SEEK_CUR if offset >= 0:
+                self.read(offset) # advance by offset
+            case os.SEEK_END if offset == 0:
+                self.read() # advance to end of stream
+            case _:
+                raise io.UnsupportedOperation()
```

Comment:
This should have an exception message to clarify why it's being thrown.

Here's some example language:

>"SeekEmulatingReader was asked to emulate a seek operation it does not support."

Since users aren't expected to trigger it, it's okay if it's fairly technical.